### PR TITLE
Added Disable Field on Action Button in Table Widget

### LINKED ISSUE #5405: [Table] - Add ability to dynamically enable/disable an action button

### PR DESCRIPTION
**Pull request by manishkushare**
Resolves #5405

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-85zrm1pe9 Add ability to dynamically enable/disable an action button](https://app.clickup.com/t/85zrm1pe9)

File: frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+export const DisableActionButtonComponent = ({
+  label,
+  currentState,
+  index,
+  darkMode,
+  callbackFunction,
+  property,
+  action = {},
+}) => {
+  const [forceCodeBox, setForceCodeBox] = React.useState(action.forceCodeBox ?? true);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx:15]:
```suggestion
  const [forceCodeBox, setForceCodeBox] = useState(action.forceCodeBox ?? true);
```

File: frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { CodeHinter } from '../../../CodeBuilder/CodeHinter';
+import FxButton from '../../../CodeBuilder/Elements/FxButton';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx:3]:
```suggestion
import { CodeHinter } from '@/Editor/CodeBuilder/CodeHinter';
import FxButton from '@/Editor/CodeBuilder/Elements/FxButton';
```

File: frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const [disabled, setDisabled] = useState(action.disableActionButton ?? false);
+  const codeHinterValue = useRef(action.codeHinterValue ?? null);
+
+  useEffect(() => {
+    callbackFunction(index, 'forceCodeBox', forceCodeBox);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [forceCodeBox]);
+
+  return (
+    <>
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx:25]:
```suggestion
```

Fragment is not needed here

File: frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      <div className="field mb-3">
+        <div className="d-flex justify-content-between">
+          <div>
+            <label className="form-label">{label}</label>
+          </div>
+          <div className="col-auto">
+            <FxButton
+              active={!forceCodeBox ? true : false}
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Inspector/Components/Table/DisableActionButtonConponent.jsx:33]:
```suggestion
              active={forceCodeBox ? false: true}
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@manishkushare Please check the suggested code changes. Along with that noticed the following issues,

- Tried disabling with `Fx` seems to be not working properly for me. Can you test it out?
- When I turn on Fx, the UI is different. Can we make it consistent with the current FX?
- If possible, try to import with `@` instead of `../../`. It's better readable and consistent through out the codebase

**Review by kavinvenkatachalam — APPROVED** [review]

**Comment by kavinvenkatachalam** [comment]
@manishkushare Can you resolve the conflicts?

**Comment by kavinvenkatachalam** [comment]
It causes slowness of the app as we need to pass complete currentState to the table column. So, adding this PR `On Hold`

**Review by kavinvenkatachalam — APPROVED** [review]

**Review by kavinvenkatachalam — APPROVED** [review]

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**